### PR TITLE
KLBP-29: Step 2 create new person.

### PR DESCRIPTION
**Pull request by andreir-suciu**

### CODE REVIEW

File: src/interfaces/Folder.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export interface IFolderContent {
   nom: string
   prenom: string
   telephone: string
+  is_kyb_legal_representative: string
```
Comment (ioan-iacob-netea) [inline src/interfaces/Folder.ts:15]: Please use camelCase for this property name.

File: src/enums/journeys.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const RL_PERSON_TYPE_ID = '1'
+
+export const STEP_1_ID = '0'
```
Comment (ioan-iacob-netea) [inline src/enums/journeys.ts:3]: :+1:

File: src/components/step2/Step2Form.tsx [hunk 3]
```diff
@@ -1,29 +1,45 @@
...
+const YES_VALUE = '1'
+const NO_VALUE = '0'
 
-const Step2Form: FC = () => {
+export interface IStep2HookData {
+  dispatch: Dispatch<Action>
+  form: UseFormReturn<Record<string, IStep2FormData>>
+  isLegalRepresentative: string | undefined
+  documentType: string | undefined
+  validateFiles: () => undefined | true
```
Comment (ioan-iacob-netea) [inline src/components/step2/Step2Form.tsx:18]:
```suggestion
  validateFiles: () => undefined | bool
```

**Review by ioan-iacob-netea — CHANGES_REQUESTED** [review]

File: src/__mock__/folderMock.ts [hunk 4]
```diff
@@ -6,4 +6,5 @@ export const defaultFolder: IFolderContent = {
   email: 'email',
   telephone: 'telephone',
   journeyStatus: '1',
+  is_kyb_legal_representative: '1',
```
[the remark this answers is not in the document]
Comment (andreir-suciu) [reply src/__mock__/folderMock.ts]: Done

File: src/__tests__/pages/steps/Step2.test.tsx [hunk 5]
```diff
@@ -54,17 +60,107 @@ describe('Render Step2 component', () => {
...
+
+  beforeEach(() => {
+    fetch.resetMocks()
+    store = mockedStore
+  })
+
+  test('Step 2 form initial', async () => {
+    setup(store)
+
+    expect(screen.queryByText('Nom du RL*')).not.toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (andreir-suciu) [reply src/__tests__/pages/steps/Step2.test.tsx]: Updated.

File: src/interfaces/Folder.ts (hunk 1 above)
Comment (andreir-suciu) [reply src/interfaces/Folder.ts]: Done

File: src/components/step2/Step2Form.tsx (hunk 3 above)
Comment (andreir-suciu) [reply src/components/step2/Step2Form.tsx]: reviewed all interface properties at is was automatically generated and updated where necessary

**Review by ioan-iacob-netea — APPROVED** [review]

**Review by iuliudascalu — APPROVED** [review]

**Review by ioan-iacob-netea — APPROVED** [review]

File: src/components/StepNavButtons/StepNavButtons.tsx [hunk 6]
```diff
@@ -6,6 +6,7 @@ import { IStepNavButtons } from '../../interfaces/Step'
 const StepNavButtons = ({
   displayNext = true,
   displayPrevious = true,
+  nextButtonType = 'submit',
```
Comment (mariuspop86) [inline src/components/StepNavButtons/StepNavButtons.tsx:9]: I don't think this change is really necessar

File: src/components/step2/Step2Form.tsx [hunk 7]
```diff
@@ -1,29 +1,46 @@
...
+import { IPersonData, IStep2FormData } from '../../interfaces/Step2'
+import { IFileState } from '../../interfaces/Upload'
 
-const YES_VALUE = '0'
-const NO_VALUE = '1'
+const YES_VALUE = '1'
+const NO_VALUE = '0'
 
-const Step2Form: FC = () => {
+export interface IStep2HookData {
```
Comment (mariuspop86) [inline src/components/step2/Step2Form.tsx:14]: Please move this to the interfaces folder

File: src/components/step2/Step2Form.tsx [hunk 8]
```diff
@@ -1,29 +1,46 @@
...
 import DynamicFormattedMessage from '../common/DynamicFormattedMessage'
-import useStep2FormHook from '../../hooks/useStep2FormHook'
-import StepNavButtons from '../StepNavButtons/StepNavButtons'
 import LegalRepresentativeSection from '../LegalRepresentativeSection/LegalRepresentativeSection'
+import { IPersonData, IStep2FormData } from '../../interfaces/Step2'
+import { IFileState } from '../../interfaces/Upload'
 
-const YES_VALUE = '0'
-const NO_VALUE = '1'
+const YES_VALUE = '1'
```
Comment (mariuspop86) [inline src/components/step2/Step2Form.tsx:11]: Please move this in the enums folder

File: src/hooks/useStep1FormHook.ts [hunk 9]
```diff
@@ -10,7 +10,7 @@ const initialState: IFileState[] = [
   {
     file: null,
     message: '',
-    personTypeId: '1', // TODO update when we have the correct type
+    personTypeId: '101', // TODO update when we have the correct type
```
Comment (mariuspop86) [inline src/hooks/useStep1FormHook.ts:13]: Please revert this. This was fixed here https://github.com/beprems/KYB-lbp-front/pull/60

File: src/interfaces/FormFields.ts [hunk 10]
```diff
@@ -64,15 +64,16 @@ export interface ISelectOption {
...
+  error?: FieldError
   options: ISelectOption[]
+  control: Control<Record<string, any>>
+  onChange?: (value: string) => void
+  defaultValue?: ISelectOption
+  errorClassName?: string
   page: string
+  hasLabel?: boolean
   placeholder?: string
+  rules?: any
```
Comment (mariuspop86) [inline src/interfaces/FormFields.ts:78]:
```suggestion
  rules?: Record<string, object>
```

File: src/interfaces/LegalRepresentative.ts [hunk 11]
```diff
@@ -16,20 +17,23 @@ export interface ILegalRepresentativeCard {
...
 }
 
 export interface ISelectUploadDocument extends ILegalRepresentative {
   id: string
-  hasLabel?: boolean
   documentLabel: { [key: string]: string }
   options: IOption[]
   className?: string
+  hasLabel?: boolean
+  rules?: any
```
Comment (mariuspop86) [inline src/interfaces/LegalRepresentative.ts:38]: Here also.

File: src/hooks/useStep2Hook.ts [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+  const onClickNext = async (): Promise<void> => {
+    dispatch(folderActions.folderLoading())
+    const formData: IFormData = {
+      data: {
+        isKybLegalRepresentative: isLegalRepresentative,
+      },
+      step,
+      next_step: step + 1,
+    }
+
```
Comment (mariuspop86) [inline src/hooks/useStep2Hook.ts:130]:
```suggestion
onSubmit(formData)
```

**Review by mariuspop86 — APPROVED** [review]

File: src/components/StepNavButtons/StepNavButtons.tsx (hunk 6 above)
Comment (andreir-suciu) [reply src/components/StepNavButtons/StepNavButtons.tsx]: Removed nextButtonType.

File: src/components/step2/Step2Form.tsx (hunk 7 above)
Comment (andreir-suciu) [reply src/components/step2/Step2Form.tsx]: Moved to Step2.ts

File: src/components/step2/Step2Form.tsx (hunk 8 above)
Comment (andreir-suciu) [reply src/components/step2/Step2Form.tsx]: Moved to enum folders.

File: src/hooks/useStep1FormHook.ts (hunk 9 above)
Comment (andreir-suciu) [reply src/hooks/useStep1FormHook.ts]: Done

File: src/interfaces/FormFields.ts (hunk 10 above)
Comment (andreir-suciu) [reply src/interfaces/FormFields.ts]: Fixed.

File: src/interfaces/LegalRepresentative.ts (hunk 11 above)
Comment (andreir-suciu) [reply src/interfaces/LegalRepresentative.ts]: Fixed.

File: src/hooks/useStep2Hook.ts (hunk 12 above)
Comment (andreir-suciu) [reply src/hooks/useStep2Hook.ts]: Fixed.

File: src/hooks/useStep2Hook.ts [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+      getPersonData()
+    } catch (error) {
+      handleError(error, history)
+      dispatch({
+        type: UPLOAD_ACTIONS.setStatus,
+        payload: { index: 0, status: UPLOAD_STATUS.idle },
+      })
+
+      if (isUnauthorized(error)) {
```
Comment (mariuspop86) [inline src/hooks/useStep2Hook.ts:110]: This is not needed, it's already done in `handleError`

Comment (andreir-suciu) [reply src/hooks/useStep2Hook.ts]: Removed.

**Pull request merged**